### PR TITLE
improve stack doc and fix some dependencies in terraform

### PR DESCRIPTION
**Pull request by gaelL**

### CODE REVIEW

File: terraform/infrastructure.tf.sample [hunk 1]
```diff
@@ -10,36 +10,111 @@ module "infrastructure" {
...
+  # staging VPC
+  #
+
+  #. staging_cidr: 10.1.0.0/16
+  #+ The CIDR of the staging VPC
+  staging_cidr                = "10.1.0.0/16"
+
+  #. staging_private_subnets (optional, list): ["10.1.0.0/24", "10.1.2.0/24", "10.1.4.0/24"]
+  #+ The private subnets for the staging VPC
+  staging_private_subnets     = ["10.1.0.0/24", "10.1.4.0/24", "10.1.8.0/24"]
```
Comment (sdurrheimer) [inline terraform/infrastructure.tf.sample:56]:
```suggestion
  staging_private_subnets     = ["10.1.0.0/24", "10.1.2.0/24", "10.1.4.0/24"]
```

File: terraform/infrastructure.tf.sample [hunk 2]
```diff
@@ -10,36 +10,111 @@ module "infrastructure" {
...
+
+  #. staging_public_subnets (optional, list): ["10.1.1.0/24", "10.1.3.0/24", "10.1.5.0/24"]
+  #+ The public subnets for the staging VPC
+  staging_public_subnets      = ["10.1.1.0/24", "10.1.5.0/24", "10.1.9.0/24"]
```
Comment (sdurrheimer) [inline terraform/infrastructure.tf.sample:60]:
```suggestion
  staging_public_subnets      = ["10.1.1.0/24", "10.1.3.0/24", "10.1.5.0/24"]
```

File: terraform/infrastructure.tf.sample [hunk 3]
```diff
@@ -10,36 +10,111 @@ module "infrastructure" {
...
+
+  #. staging_rds_subnets (optional, list): []
+  #+ The RDS subnets for the staging VPC
+  staging_rds_subnets         = ["10.1.2.0/24", "10.1.6.0/24", "10.1.10.0/24"]
```
Comment (sdurrheimer) [inline terraform/infrastructure.tf.sample:64]:
```suggestion
  staging_rds_subnets         = []
```

File: terraform/infrastructure.tf.sample [hunk 4]
```diff
@@ -10,36 +10,111 @@ module "infrastructure" {
...
+
+  #. staging_elasticache_subnets (optional, list): []
+  #+ The Elasticache subnets for the staging VPC
+  staging_elasticache_subnets = ["10.1.3.0/24", "10.1.7.0/24", "10.1.11.0/24"]
```
Comment (sdurrheimer) [inline terraform/infrastructure.tf.sample]:
```suggestion
  staging_elasticache_subnets = []
```

File: terraform/infrastructure.tf.sample [hunk 5]
```diff
@@ -10,36 +10,111 @@ module "infrastructure" {
...
+  # prod VPC
+  #
+
+  #. prod_cidr: 10.2.0.0/16
+  #+ The CIDR of the prod VPC
+  prod_cidr                = "10.2.0.0/16"
+
+  #. prod_private_subnets (optional, list): ["10.2.0.0/24", "10.2.2.0/24", "10.2.4.0/24"]
+  #+ The private subnets for the prod VPC
+  prod_private_subnets     = ["10.2.0.0/24", "10.2.4.0/24", "10.2.8.0/24"]
```
Comment (sdurrheimer) [inline terraform/infrastructure.tf.sample:79]:
```suggestion
  prod_private_subnets     = ["10.2.0.0/24", "10.2.2.0/24", "10.2.4.0/24"]
```

File: terraform/infrastructure.tf.sample [hunk 6]
```diff
@@ -10,36 +10,111 @@ module "infrastructure" {
...
+
+  #. prod_public_subnets (optional, list): ["10.2.1.0/24", "10.2.3.0/24", "10.2.5.0/24"]
+  #+ The public subnets for the prod VPC
+  prod_public_subnets      = ["10.2.1.0/24", "10.2.5.0/24", "10.2.9.0/24"]
```
Comment (sdurrheimer) [inline terraform/infrastructure.tf.sample:83]:
```suggestion
  prod_public_subnets      = ["10.2.1.0/24", "10.2.3.0/24", "10.2.5.0/24"]
```

File: terraform/infrastructure.tf.sample [hunk 7]
```diff
@@ -10,36 +10,111 @@ module "infrastructure" {
...
+
+  #. prod_rds_subnets (optional, list): []
+  #+ The RDS subnets for the prod VPC
+  prod_rds_subnets         = ["10.2.2.0/24", "10.2.6.0/24", "10.2.10.0/24"]
```
Comment (sdurrheimer) [inline terraform/infrastructure.tf.sample:87]:
```suggestion
  prod_rds_subnets         = []
```

File: terraform/infrastructure.tf.sample [hunk 8]
```diff
@@ -10,36 +10,111 @@ module "infrastructure" {
...
+
+  #. prod_elasticache_subnets (optional, list): []
+  #+ The Elasticache subnets for the prod VPC
+  prod_elasticache_subnets = ["10.2.3.0/24", "10.2.7.0/24", "10.2.11.0/24"]
```
Comment (sdurrheimer) [inline terraform/infrastructure.tf.sample]:
```suggestion
  prod_elasticache_subnets = []
```

File: README.md [hunk 9]
```diff
@@ -4,35 +4,151 @@ Service catalog infrastructure stack
...
 
-  * AWS S3 bucket to upload backups
-  * AWS S3 bucket to upload deployment artefacts
-  * AWS SSH key pair
-  * AWS EC2 bastion server
-  * AWS VPCs and subnets
-  * AWS SES service to allow your EC2 servers send system mails
-  * AWS IAM roles and admin policies
+<p align="center">
+<img src="https://raw.githubusercontent.com/cycloid-community-catalog/stack-infrastructure/master/diagram.jpeg" width="400">
```
Comment (adriagalin) [inline README.md]: we can change this to:

``

or does it make sense if we create a `_docs` folder with all this kind of stuff inside?

File: README.md [hunk 10]
```diff
@@ -4,35 +4,151 @@ Service catalog infrastructure stack
...
+> That means if you trigger a terraform plan, to apply it, you have to go on terraform apply job
+> and click on the `+` button to trigger it.
+
+<img src="https://raw.githubusercontent.com/cycloid-community-catalog/stack-infrastructure/master/pipeline.png" width="800">
+
+**Jobs description**
+
+  * `terraform-plan`: Terraform job that will simply make a plan of the infrastructure's stack.
+  * `terraform-apply`: Terraform job similar to the plan one, but will actually create/update everything that needs to. Please see the plan diff for a better understanding.
+  * `terraform-destroy`: **/!\ Destroy /!\** Terraform job meant to destroy the whole stack - **NO CONFIRMATION ASKED**. If triggered, the full project **WILL** be destroyed. Use with caution.
```
Comment (adriagalin) [inline README.md]: we can use :warning: emoji instead of /!\

File: README.md (hunk 9 above)
Comment (gaelL) [reply README.md]: nice catch for the url, I will change it everywere.

for the docs directory why not let's move all images in it

Comment (adriagalin) [reply README.md]: 100% agree

**Review by sdurrheimer — APPROVED** [review]

**Review by adriagalin — APPROVED** [review]

**Pull request merged**